### ISSUE TITLE
Occasionally, ContentId.generate().toAddress() returns binary data

### ISSUE THREAD

**Issue by jfinkhaeuser** [Joystream/apps#208]
In my code, I'm doing this:

```javascript
    const { ContentId } = require('@joystream/types/lib/media');
    var cid = ContentId.generate();
    cid = cid.toAddress();
    console.log(cid);
```

Occasionally, this does not output base58-encoded data as expected, but binary data. I suspect that `toAddress()` receives some data it cannot encode, but I have no further clue so far.

**Comment by siman** [comment]
you need `cid.toAddress().toString()`

**Comment by jfinkhaeuser** [comment]
Doesn't seem to be enough:

```
Generated content ID: �.���/m�8��D�-�zpP���(@��q]�
```

**Comment by jfinkhaeuser** [comment]
Isolated this error.

**Output**

```bash
$ node test.js 
Uint8Array [
  144,
  151,
  134,
  226,
  174,
  192,
  122,
  204,
  59,
  45,
  59,
  16,
  214,
  145,
  151,
  194,
  201,
  42,
  207,
  211,
  220,
  242,
  63,
  205,
  245,
  132,
  235,
  24,
  218,
  85,
  88,
  41 ]
9
u8aToString �����z�;-;֑���*����?�����UX)
encodeAddress 5FLHmMDoJ6nLLDbtfv3THMzTfqjbhvvNRxWmu8vTYMVAzA76
```

**Code**
```javascript
const { randomAsU8a } = require('@polkadot/util-crypto');
const { encodeAddress } = require('@polkadot/keyring');
const { u8aToString, stringToU8a } = require('@polkadot/util');

const random = randomAsU8a();
console.log(random);
console.log(random.indexOf('-'.charCodeAt(0)));

console.log('u8aToString', u8aToString(random));
console.log('encodeAddress', encodeAddress(random));
```

If the random Bytes contain a `-` character, `encodeAddress` is not used, and the resulting string contains all kinds of binary data.

**Comment by jfinkhaeuser** [comment]
I guess for Acropolis, we can kill this test for UUIDs already - so that should resolve the error I'm seeing.

**Comment by siman** [comment]
Yes, we can remove that UUID-backward-compatibility thing. 
What are we going to do with already uploaded content?

**Comment by jfinkhaeuser** [comment]
If we want to keep it, then this is a bug that needs fixing. If we want to ditch it, we need to remove this. Either way, the current code is unfortunately broken.

**Comment by mnaamani** [comment]
Fixed in #204
